resizer.go

```diff
@@ -256,7 +256,9 @@ func extractOrEmbedImage(image *C.VipsImage, o Options) (*C.VipsImage, error) {
 
 	switch {
 	case o.Gravity == GravitySmart, o.SmartCrop:
-		image, err = vipsSmartCrop(image, o.Width, o.Height)
+		if inWidth > o.Width || inHeight > o.Height {
```

Comment:
This seems correct!

However this check appears to apply to many, if not all, of the cases in this function. So perhaps we should generalise this check before the switch statement? I'm not intimately familiar with bimg or libvips regarding handling insensible parameters. But I suspect a Crop, Embed, Trim or Extract wouldn't make sense if the input is smaller than any of the desired areas.

However, I'm just speculating, I actually don't know.

@pohang Did you already try that perhaps?

Reply:
I'll check the other cases for errors. At the time, I only checked smartcrop and center crop. The other crop case is fine due to the min/max logic.
```
width := int(math.Min(float64(inWidth), float64(o.Width)))
height := int(math.Min(float64(inHeight), float64(o.Height)))
left, top := calculateCrop(inWidth, inHeight, o.Width, o.Height, o.Gravity)
left, top = int(math.Max(float64(left), 0)), int(math.Max(float64(top), 0))
```

Reply:
It seems like only smartcrop has this issue. Crop has special logic to avoid the bad extract area. The check can still be used to avoid calling vips to crop an image to the exact same size.

For embed, it makes sense to embed a smaller picture to a larger one. Trim is not concerned with the output size. Extract seems like user error if they input something insensible.